services/StockService.php

```diff
@@ -1048,15 +1048,20 @@ public function OpenProduct(int $productId, float $amount, $specificStockEntryId
 				]);
 				$logRow->save();
 
-				$stockEntry->update([
+				$update = [
 					'amount' => $amount,
 					'open' => 1,
 					'opened_date' => date('Y-m-d'),
 					'best_before_date' => $newBestBeforeDate
-				]);
+				];
 
 				$amount = 0;
 			}
+			if ($product->move_on_open) {
+				$update['location_id'] = $product->default_consume_location_id ?? $stockEntry->location_id;
+			}
+
+			$stockEntry->update($update);
```

Comment:
To keep it consistent, transfer from/to stock journal entries should also be created when the product is "moved on open" (just like when the transfer transaction would have been done manually).

Reply:
Thinking about it, should I flip the logic and check first, and then set the `stock_log` straight to the new location if there is one?

Reply:
Read some more code: nope!

Reply:
A normal/manual transfer action creates two stock journal entries, the negative amount at the "from location" and the positive amount at the "to location":

![grafik](https://user-images.githubusercontent.com/2161815/163789918-03954ba6-e177-4c92-9f2d-e2b8c43902d6.png)

Is done by [`StockService->TransferProduct`](https://github.com/grocy/grocy/blob/0152f1c69d3cb2bb3ceba668aa1006caa19a91c6/services/StockService.php#L1180) maybe just call that function there...

Reply:
Thanks for the pointer, it makes much more sense to use the existing function.